lib/ftp.c

```diff
@@ -2067,6 +2067,31 @@ static bool ftp_213_date(const char *p, int *year, int *month, int *day,
   return TRUE;
 }
 
+static CURLcode client_write_header(struct Curl_easy *data,
+                                    char *buf, size_t blen)
+{
+  /* Some replies from an FTP server are written to the client
+   * as CLIENTWRITE_HEADER, formatted as if they came from a
+   * HTTP conversation.
+   * In all protocols, CLIENTWRITE_HEADER data is only passed to
+   * the body write callback when data->set.include_header is set
+   * via CURLOPT_HEADER.
+   * For historic reasons, FTP never played this game and expects
+   * all its HEADERs to do that always. Set that flag during the
+   * call to Curl_client_write() so it does the right thing.
+   *
+   * Notice that we cannot enable this flag for FTP in general,
+   * as an FTP transfer might involve a HTTP proxy connection and
+   * headers from CONNECT should not automatically be part of the
+   * output. */
+  CURLcode result;
+  int save = data->set.include_header;
+  data->set.include_header = TRUE;
```

Comment:
Where within `Curl_client_write()` is `data->set.include_header` checked? I can't find it, so I'm not following why it needs to be set like this!

Reply:
Never mind. I see it now, this patch adds the check.

Would it not be nicer to pass in the "include_header" bool as a new argument to Curl_client_write  instead of changing the variable like this?

Reply:
> Never mind. I see it now, this patch adds the check.
> 
> Would it not be nicer to pass in the "include_header" bool as a new argument to Curl_client_write instead of changing the variable like this?

My thinking is that FTP is the sole exception here and that every other protocol does not have to care about this. I'd rather spare all others the "burden" to think about it than make FTP look nicer.

Reply:
Ah right. Sounds fair.